tests/python/pants_test/invalidation/test_cache_manager.py

```diff
@@ -56,8 +58,33 @@ def tearDown(self):
     shutil.rmtree(self._dir, ignore_errors=True)
     super(InvalidationCacheManagerTest, self).tearDown()
 
-  def make_vts(self, target):
-    return VersionedTarget(self.cache_manager, target, target.id)
+  def make_vt(self, invalid=False):
+    # Create an arbitrary VT. If invalid is False, it will mimic the state of the VT handed back by a task.
+    a_target = self.make_target(':a', dependencies=[])
+    ic = self.cache_manager.check([a_target])
+    vt = ic.all_vts[0]
+    if not invalid:
+      self.task_execute(vt)
+      vt.update()
+    return vt
+
+  def task_execute(self, vt):
+    vt.create_results_dir(self._dir)
+    task_output = os.path.join(vt.results_dir, 'a_file')
+    self.create_file(task_output, 'foo')
+
+  def is_empty(self, dirname):
+    return not os.listdir(dirname)
+
+  def matching_result_dirs(self, vt):
+    # Ensure that the result_dirs contain the same files.
+    return self.is_empty(vt.results_dir) == self.is_empty(vt.current_results_dir)
```

Comment:
Yeah, this for sure needs to be changed. I was continually testing the file list and decided to break it out into a function - but I did not do a very good job.